File: back-end/utils/logger.py

class Logger:
   
    colors = {
        'error': '\033[91m',     # red color
        'warning': '\033[93m',   # yellow color
        'info': '\033[94m',       # blue color
        'debug': '\033[94m'       # blue color
    }
    end_color = '\033[0m'   # reset color
    is_debug = False
    
    @classmethod
    def debug(cls, *messages):
        if not cls.is_debug: 
            return 
        
        level = "debug"
        color = cls.colors[level]
        formatted_messages = ' '.join(str(message) for message in messages)
        log_message = f"{color}[{level.upper()}]{cls.end_color} {formatted_messages}"
        print(log_message)
    
    @classmethod
    def info(cls, *messages):
        level = "info"
        color = cls.colors[level]
        formatted_messages = ' '.join(str(message) for message in messages)
        log_message = f"{color}[{level.upper()}]{cls.end_color} {formatted_messages}"
        print(log_message)
    
    
    @classmethod
    def warn(cls, *messages):
        level = "warning"
        color = cls.colors[level]
        formatted_messages = ' '.join(str(message) for message in messages)
        log_message = f"{color}[{level.upper()}]{cls.end_color} {formatted_messages}"
        print(log_message)
    
    
    @classmethod
    def error(cls, *messages):
        level = "error"
        color = cls.colors[level]
        formatted_messages = ' '.join(str(message) for message in messages)
        log_message = f"{color}[{level.upper()}]{cls.end_color} {formatted_messages}"
        print(log_message)

File: client-side/src/utils/logger.js
import chalk from "chalk";
import { chain } from "lodash";

const error = chalk.bold.red;
const warning = chalk.bold.hex("#FFA500");
const info = chalk.bold.green;

const log = (type, msg) => {
  const logMessage = chain(msg)
    .map((arg) => (typeof arg === "object" ? JSON.stringify(arg) : arg))
    .join("")
    .value();
  console.log(`${type}: `, logMessage);
};

export const logger = {
  info: (...msg) => {
    log(info("[Info]"), msg);
  },
  warn: (...msg) => {
    log(warning("[Warn]"), msg);
  },
  error: (...msg) => {
    log(error("[Error]"), msg);
  },
};
